server/lib/middleware.js

```diff
@@ -85,18 +90,37 @@ module.exports = function middlewareConstructor(config) {
       let token = req.user.token = cryptr.decrypt(req.session.token);
 
       if(ASSERT_TOKEN) {
-        // We expect a 64 character HEX string for the token, for example:
-        // '16d4c4d2fa4d6e4aa6b1b5c6a115ad44fd271dd98204f2008bf5efbba5a56dec'
-        let tokenType = typeof token;
-        if(!token) {
-          console.log("ASSERT_TOKEN FAILED: Expected token to exist");
-        } else {
+        /**
+         * Assert whether or not this token is correct or rather broken.
+         * We expect a 64 character HEX string for the token, for example:
+         * '16d4c4d2fa4d6e4aa6b1b5c6a115ad44fd271dd98204f2008bf5efbba5a56dec'
+         */
+        let assert = function(token) {
+          if(!token) {
+            console.log("ASSERT_TOKEN FAILED: Expected token to exist");
+            return false;
+          }
+
+          let tokenType = typeof token;
+          
           if(tokenType !== "string") {
             console.log("ASSERT_TOKEN FAILED: Expected token type to be String, instead got: " + tokenType);
-          } else {
-            if(!/^[a-z0-9]{64}$/.test(token)) {
-              console.log("ASSERT_TOKEN FAILED: Expected token to only have chars a-z, 0-9. Also got: '" +  token.replace(/[a-z0-9]/g, ' ') + "'");
-            }
+            return false;
+          }
+          
+          if(!/^[a-z0-9]{64}$/.test(token)) {
+            console.log("ASSERT_TOKEN FAILED: Expected token to only have chars a-z, 0-9. Also got: '" +  token.replace(/[a-z0-9]/g, ' ') + "'");
+            return false;
+          }
+          
+          return true;
+        };
+
+        if (!assert(token)) {
+          console.log("ASSERT_TOKEN FAILED: retrying decryption using aes-256 rather than aes-256-ctr");
+          token = req.user.token = cryptrFallback.decrypt(req.session.token);
+          if (!assert(token)) {
+            return next(new Error("Session token cannot be decrypted"));
```

Comment:
Can you update this string to say, `"Session token cannot be decrypted. Please sign out and sign in again."` so that there is some info on what to do?